docs/manual/java/guide/cluster/UsingAkkaPersistenceTyped.md

```diff
@@ -0,0 +1,185 @@
+
+So we declare an implicit `timeout` and then invoke `ask` (which uses the timeout implicitly). The `ask` method accepts a function of `ActorRef[Res] => M` in which `Res` is the expected response type and `M` is the message being sent to the actor. The `ask` method will create an instance of `ActorRef[Res]` that can be used to build the outgoing message (command). Once the response is sent to `ActorRef[Res]`, Akka will complete the returned `CompletionStage<Res>` with the response (in this case `CompletionStage[ShoppingCartSummary]`).
+
+Finally, we operate over the `cartSummary` (in this case, we map it to a different type, ie: `ShoppingCartView`, using the `thenApply` method).
+
+The `ShoppingCartView` and `asShoppingCartView` are defined as:
+
+@[shopping-cart-service-view](code/docs/home/persistence/ShoppingCartServiceImpl.java)
+
+@[shopping-cart-service-map](code/docs/home/persistence/ShoppingCartServiceImpl.java)
+
+> **Note**: We are keeping the internal state of the Aggregate isolated from the exposed service API so they can evolve independently.
+
+### Configuring number of shards
+
+Akka recommends, as a rule of thumb, that the number of shards should be a factor ten higher than the planned maximum number of cluster nodes. It doesn't have to be exact. Fewer shards than the number of nodes will result in that some nodes will not host any shards. Too many shards will result in less efficient management of the shards, e.g. rebalancing overhead, and increased latency because the coordinator is involved in the routing of the first message for each shard.
+
+See the [Akka Cluster Sharding documentation](https://doc.akka.io/docs/akka/2.6/typed/cluster-sharding.html?language=Java#shard-allocation) for details on how to configure the number of shards.
+
+### Configuring Entity passivation
+
+Keeping all the Aggregates in memory all the time is inefficient. Instead, use the Entity passivation feature, then sharded entities (the Aggregates) are removed from the cluster when they've been unused for some time.
+
+Akka supports both [programmatic passivation](https://doc.akka.io/docs/akka/2.6/typed/cluster-sharding.html?language=Java#passivation) and [automatic passivation](https://doc.akka.io/docs/akka/2.6/typed/cluster-sharding.html?language=Java#automatic-passivation). The default values for automatic passivation are generally good enough.
+
+## Data Serialization
+
+The messages (commands, replies) and the durable classes (events, state snapshots) need to be serializable to be sent over the wire across the cluster or be stored on the database. Akka recommends [Jackson-based serializers](https://doc.akka.io/docs/akka/2.6/serialization-jackson.html?language=Java) --preferably JSON, but CBOR is also supported-- as a good default in most cases. On top of Akka serializers, Lagom makes it easy to add [[Play-JSON serialization|SerializationPlayJson]] support, which may be more familiar to some Scala developers.
```

Comment:
Is this Play-JSON discussion relevant to Java users? They will use Jackson for everything, so I think this can be removed from here.

Reply:
The goal was to remove it. I removed a few paragraphs, but missed this sentence.